### ISSUE TITLE
CI: Doc build is failing because of upstream changes

### ISSUE THREAD

**Issue by phofl** [pandas-dev/pandas#44494]
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
>>>-------------------------------------------------------------------------
Warning in /home/runner/work/pandas/pandas/doc/source/user_guide/io.rst at block ending on line 3033
Specify :okwarning: as an option in the ipython:: block to suppress this message
----------------------------------------------------------------------------
/home/runner/miniconda3/envs/pandas-dev/lib/python3.8/site-packages/charset_normalizer/api.py:91: UserWarning: Trying to detect encoding from a tiny portion of (11) byte(s).
  warn('Trying to detect encoding from a tiny portion of ({}) byte(s).'.format(length))
<<<-------------------------------------------------------------------------

and

WARNING: failed to reach any of the inventories with the following issues:
intersphinx inventory 'https://www.statsmodels.org/devel/objects.inv' not fetchable due to <class 'requests.exceptions.ConnectionError'>: HTTPSConnectionPool(host='www.statsmodels.org', port=443): Max retries exceeded with url: /devel/objects.inv (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fef8dc72670>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))
```

**Comment by phofl** [comment]
cc @pandas-dev/pandas-core

**Comment by bashtage** [comment]
Seems to be available now.

**Comment by johnzangwill** [comment]
This is the log from this morning:
[error_log.log](https://github.com/pandas-dev/pandas/files/7554285/error_log.log)

**Comment by bashtage** [comment]
Looks like he `NumericIndex` changes are biting from io.rst.

**Comment by jbrockmendel** [comment]
Any thoughts on how to handle these warnings?  my guess of catching warnings in conf.linkcode_resolve was a bust

**Comment by lithomas1** [comment]
Looks like this fixed itself?

**Comment by jbrockmendel** [comment]
closed by #44529

**Comment by jreback** [comment]
yep
